--- a/gpio/OrangePi.h	1970-01-01 03:00:00.000000000 +0300
+++ b/gpio/OrangePi.h	2022-02-01 22:33:59.693390921 +0200
@@ -0,0 +1,540 @@
+
+#ifdef CONFIG_ORANGEPI_ZERO
+static char *physNames [64] = 
+{
+      NULL,
+
+ "   +3.3v", "     +5v",
+ "PA12 sda", "     +5v",
+ "PA11 scl", "     GND",
+ "PA6     ", " PG6    ",
+ "     GND", " PG7    ",
+ "PA1     ", " PA7    ",
+ "PA0     ", "     GND",
+ "PA3     ", " PA19   ",
+ "   +3.3v", " PA18   ",
+ "PA15    ", "     GND",
+ "PA16    ", " PA2    ",
+ "PA14    ", " PA13   ",
+ "     GND", " PA10   ",
+
+ "RED-LED ", "uart-PWM",
+ 
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL,
+};
+#endif /* ORANGEPI_ZERO_H2+ */
+
+#ifdef CONFIG_ORANGEPI_PC2
+char *physNames [384] =
+{
+    NULL,
+
+ "    3.3V", "5V      ",
+ "   SDA.0", "5V      ",
+ "   SCL.0", "GND     ",
+ "   PWM.1", "PC05    ",
+ "     GND", "PC06    ",
+ "   RXD.2", "PD14    ",
+ "   TXD.2", "GND     ",
+ "   CTS.2", "PC04    ",
+ "    3.3V", "PC07    ",
+ "  MOSI.1", "GND     ",
+ "  MISO.1", "RTS.2   ",
+ "  SCLK.1", "CE.1    ",
+ "     GND", "PA21    ",
+ "   SDA.1", "SCL.1   ",
+ "    PA07", "GND     ",
+ "    PA08", "RTS.1   ",
+ "    PA09", "GND     ",
+ "    PA10", "CTS.1   ",
+ "    PD11", "TXD.1   ",
+ "     GND", "RXD.1   ",
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+   NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,
+};
+#endif /* PC2 */
+
+#ifdef CONFIG_ORANGEPI_PRIME
+char *physNames [384] =
+{
+    NULL,
+
+ "    3.3V", "5V      ",
+ "   SDA.0", "5V      ",
+ "   SCL.0", "GND     ",
+ "   PWM.1", "PC05    ",
+ "     GND", "PC06    ",
+ "   RXD.2", "PD14    ",
+ "   TXD.2", "GND     ",
+ "   CTS.2", "PC04    ",
+ "    3.3V", "PC07    ",
+ "  MOSI.1", "GND     ",
+ "  MISO.1", "RTS.2   ",
+ "  SCLK.1", "CE.1    ",
+ "     GND", "PC08    ",
+ "   SDA.1", "SCL.1   ",
+ "    PA07", "GND     ",
+ "    PA08", "PC09    ",
+ "    PA09", "GND     ",
+ "    PA10", "PC10    ",
+ "    PD11", "PC11    ",
+ "     GND", "PC12    ",
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+   NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,
+};
+#endif /* PRIME */
+
+#ifdef CONFIG_ORANGEPI_2G_IOT
+char *physNames [384] =
+{
+    NULL,
+
+ "    3.3V", "5V      ",
+ "   SDA.0", "5V      ",
+ "   SCL.0", "GND     ",
+ "  GPIO.7", "TXD.2   ",
+ "     GND", "RXD.2   ",
+ "   RXD.1", "GPIO.1  ",
+ "   TXD.1", "GND     ",
+ "   CTS.1", "GPIO.4  ",
+ "    3.3V", "GPIO.5  ",
+ "   SDI.2", "GND     ",
+ "  SDIO.2", "RTS1    ",
+ "  SCLK.2", "CE.0    ",
+ "     GND", "CE.1    ",
+ "   SDA.1", "SCL.1   ",
+ " GPIO.21", "GND     ",
+ " GPIO.22", "RTS.2   ",
+ " GPIO.23", "GND     ",
+ " GPIO.24", "CTS.2   ",
+ " GPIO.25", "SCL.2   ",
+ "     GND", "SDA.2   ",
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+   NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,
+};
+#endif	/* CONFIG_ORANGEPI_2G_IOT */
+
+#ifdef CONFIG_ORANGEPI_WIN
+char *physNames [384] =
+{
+  NULL,
+
+ "    3.3V", "5V      ",
+ "   SDA.1", "5V      ",
+ "   SCL.1", "GND     ",
+ "    PL10", "PL02    ",
+ "     GND", "PL03    ",
+ "   RXD.3", "PD04    ",
+ "   TXD.3", "GND     ",
+ "   CTS.3", "PL09    ",
+ "    3.3V", "PC04    ",
+ "  MOSI.1", "GND     ",
+ "  MISO.1", "RTS.3   ",
+ "  SCLK.1", "CE.1    ",
+ "     GND", "PD06    ",
+ "   SDA.2", "SCL.2   ",
+ "    PB04", "GND     ",
+ "    PB05", "RTS.2   ",
+ "    PB06", "GND     ",
+ "    PB07", "CTS.2   ",
+ "    PD05", "TXD.2   ",
+ "     GND", "RXD.2   ",
+
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+   NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,
+};
+#endif /* CONFIG_ORANGEPI_WIN */
+
+#ifdef CONFIG_ORANGEPI_H3
+char *physNames [384] =
+{
+          NULL,
+
+ "    3.3V", "5V      ",
+ "   SDA.0", "5V      ",
+ "   SCL.0", "GND     ",
+ "     PA6", "TXD.3   ",
+ "     GND", "RXD.3   ",
+ "   RXD.2", "PD14    ",
+ "   TXD.2", "GND     ",
+ "   CTS.2", "PC04    ",
+ "    3.3V", "PC07    ",
+ "  MOSI.0", "GND     ",
+ "  MISO.0", "RTS.2   ",
+ "  SCLK.0", "CE.0    ",
+ "     GND", "PA21    ",
+ "   SDA.1", "SCL.1   ",
+ "    PA07", "GND     ",
+ "    PA08", "RTS.1   ",
+ "    PA09", "GND     ",
+ "    PA10", "CTS.1   ",
+ "    PA20", "TXD.1   ",
+ "     GND", "RXD.1   ",
+ "    PA04", "PA05    ",
+
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL, 
+       NULL, NULL,
+       NULL, NULL,
+	  NULL,
+};
+#endif /* CONFIG_ORANGEPI_H3 */
+
+#if CONFIG_ORANGEPI_ZEROPLUS2_H3 || CONFIG_ORANGEPI_ZEROPLUS2_H5
+char *physNames [384] =
+{
+      NULL,
+
+  "    3.3V", "5V      ",
+  "   SDA.0", "5V      ",
+  "   SCL.0", "GND     ",
+  "     PA6", "TXD.2   ",
+  "     GND", "RXD.2   ",
+  "   S-SCL", "PD11    ",
+  "   S-SDA", "GND     ",
+  "   CTS.2", "SDA.1   ",
+  "    3.3V", "SCL.1   ",
+  "  MOSI.1", "GND     ",
+  "  MISO.1", "RTS.2   ",
+  "  SCLK.1", "CE.1    ",
+  "     GND", "PD14    ",
+
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL,
+};
+#endif /* CONFIG_ORANGEPI_ZEROPLUS2_H3 - CONFIG_ORANGEPI_ZEROPLUS2_H5 */
+
+
+#ifdef CONFIG_ORANGEPI_LITE2
+char *physNames [384] = 
+{
+  	   NULL,
+
+ "    3.3V", "5V      ",
+ "   SDA.1", "5V      ",
+ "   SCL.1", "GND     ",
+ "     PH4", "PD21    ",
+ "     GND", "PD22    ",
+ "   RXD.3", "PC9     ",
+ "   TXD.3", "GND     ",
+ "   CTS.3", "PC8     ",
+ "    3.3V", "PC7     ",
+ "  MOSI.0", "GND     ",
+ "  MISO.0", "RTS.3   ",
+ "  SCLK.0", "CE.0    ",
+ "     GND", "PH3     ",
+
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL,
+};
+#endif	/* CONFIG_ORANGEPI_LITE2 */
+
+#ifdef CONFIG_ORANGEPI_3
+char *physNames [384] = 
+{
+  	   NULL,
+
+ "    3.3V", "5V      ",
+ "   SDA.0", "5V      ",
+ "   SCL.0", "GND     ",
+ "   PWM.0", "PL02    ",
+ "     GND", "PL03    ",
+ "   RXD.3", "PD18    ",
+ "   TXD.3", "GND     ",
+ "    PL10", "PD15    ",
+ "    3.3V", "PD16    ",
+ "  MOSI.1", "GND     ",
+ "  MISO.1", "PD21    ",
+ "  SCLK.1", "CE.1    ",
+ "     GND", "PL08    ",
+
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL,
+};
+#endif	/* CONFIG_ORANGEPI_3 */
+
+#ifdef CONFIG_ORANGEPI_ZERO2
+char *physNames [384] = 
+{
+  	   NULL,
+
+ "    3.3V", "5V      ",
+ "   SDA.3", "5V      ",
+ "   SCL.3", "GND     ",
+ "     PC9", "TXD.5   ",
+ "     GND", "RXD.5   ",
+ "     PC6", "PC11    ",
+ "     PC5", "GND     ",
+ "     PC8", "PC15    ",
+ "    3.3V", "PC14    ",
+ "  MOSI.1", "GND     ",
+ "  MISO.1", "PC7     ",
+ "  SCLK.1", "CE.1    ",
+ "     GND", "PC10    ",
+ "     PC1", "        ",
+ "    PI16", "        ",
+ "     PI6", "        ",
+ "    PH10", "        ",
+
+
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL,
+};
+#endif	/* CONFIG_ORANGEPI_ZERO2 */
+
+#ifdef CONFIG_ORANGEPI_RK3399
+char *physNames [384] =
+{
+  NULL,
+  "    3.3V", "5V      ",
+  "   SDA.0", "5V      ",
+  "   SCL.0", "GND     ",
+  "   GPIO4", "Tx      ",
+  "     GND", "Rx      ",
+  "  GPIO17", "GPIO18  ",
+  "  GPIO27", "GND     ",
+  "  GPIO22", "GPIO23  ",
+  "    3.3V", "GPIO24  ",
+  "    MOSI", "GND     ",
+  "    MISO", "GPIO25  ",
+  "    SCLK", "CS0     ",
+  "     GND", "CS1     ",
+  "    DNP1", "DNP2    ",
+  "   GPIO5", "GND     ",
+  "   GPIO6", "GPIO12  ",
+  "  GPIO13", "GND     ",
+  "  GPIO19", "GPIO16  ",
+  "  GPIO26", "GPIO20  ",
+  "     GND", "GPIO21  ",
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+  	   NULL, NULL,
+       NULL, NULL,
+       NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,
+};
+#endif
+
+
+#ifdef CONFIG_ORANGEPI_800
+char *physNames [384] =
+{
+	NULL,
+  "    3.3V", "5V      ",
+  "   SDA.4", "5V      ",
+  "   SCL.4", "GND     ",
+  "    PWM1", "RXD.4   ",
+  "     GND", "TXD.4   ",
+  "   TXD.2", "GPIO4_D0",
+  "   RXD.2", "GND     ",
+  "GPIO4_C5", "SDA.2   ",
+  "    3.3V", "SCL.2   ",
+  "SPI2_TXD", "GND     ",
+  "SPI2_RXD", "GPIO4_D1",
+  "SPI2_CLK", "SPI2_CS ",
+  "     GND", "GPIO4_D2",
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,
+};
+#endif
+
+#ifdef CONFIG_ORANGEPI_4
+char *physNames [384] =
+{
+  NULL,
+  "    3.3V", "5V      ",
+  "I2C2_SDA", "5V      ",
+  "I2C2_SCL", "GND     ",
+  "    PWM1", "I2C3_SCL",
+  "     GND", "I2C3_SDA",
+  "GPIO1_A1", "GPIO1_C2",
+  "GPIO1_A3", "GND     ",
+  "GPIO2_D4", "GPIO1_C6",
+  "    3.3V", "GPIO1_C7",
+  "SPI1_TXD", "GND     ",
+  "SPI1_RXD", "GPIO1_D0",
+  "SPI1_CLK", "SPI1_CS ",
+  "     GND", "GPIO4_C5",
+  "I2C2_SDA", "I2C2_SCL",
+  " I2S0_RX", "GND     ",
+  " I2S0_TX", "I2S_CLK ",
+  "I2S0_SCK", "GND     ",
+  "I2S0_SI0", "I2S0_SO0",
+  "I2S0_SI1", "I2S0_SI2",
+  "     GND", "I2S0_SI3",
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+       NULL, NULL,
+  	   NULL, NULL,
+       NULL, NULL,
+       NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,
+};
+#endif
+
+#ifdef CONFIG_ORANGEPI_R1PLUS
+char *physNames [384] =
+{
+	NULL,
+
+	"5V      ", "GND     ",
+	"SDA.0   ", "SCK.0   ",
+	"TXD.1   ", "RXD.1   ",
+	"        ", "        ",
+	"        ", "GPIO3_C0",
+	"CTS.1   ", "RTS.1   ",
+	"GPIO2_A2", "        ",
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL, NULL,
+	NULL,NULL,NULL,NULL,NULL,
+};
+#endif
