--- a/examples/PiGlow/Makefile	2022-01-13 23:50:18.000000000 +0200
+++ b/examples/PiGlow/Makefile	2022-02-01 22:33:59.721390198 +0200
@@ -29,7 +29,7 @@
 INCLUDE	= -I/usr/local/include
 CFLAGS	= $(DEBUG) -Wall $(INCLUDE) -Winline -pipe
 
-LDFLAGS	= -L/usr/local/lib
+LDFLAGS	= -L/usr/local/lib -L../wiringPi -L../devLib
 LDLIBS    = -lwiringPi -lwiringPiDev -lpthread -lm
 
 # Should not alter anything below this line
