--- a/Makefile	2022-01-13 23:50:18.000000000 +0200
+++ b/Makefile	2022-02-01 22:35:45.902662271 +0200
@@ -1,8 +1,8 @@
 all:
 	$(MAKE) -C wiringPi $@
 	$(MAKE) -C devLib $@
-	-ln -fs libwiringPiDev.so.2.0 devLib/libwiringPiDev.so
-	-ln -fs libwiringPi.so.2.0 wiringPi/libwiringPi.so
+	-ln -fs libwiringPiDev.so.2.6 devLib/libwiringPiDev.so
+	-ln -fs libwiringPi.so.2.6 wiringPi/libwiringPi.so
 	$(MAKE) -C gpio 'INCLUDE=-I../devLib -I../wiringPi' 'LDFLAGS=-L../devLib -L../wiringPi' $@
 
 %:
