--- a/feeds/packages/libs/glib2/Makefile	2024-02-05 13:52:55.437678164 +0200
+++ b/feeds/packages/libs/glib2/Makefile	2024-02-05 14:51:16.940406739 +0200
@@ -8,12 +8,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=glib2
-PKG_VERSION:=2.74.0
+PKG_VERSION:=2.70.0
 PKG_RELEASE:=$(AUTORELEASE)
 
 PKG_SOURCE:=glib-$(PKG_VERSION).tar.xz
-PKG_SOURCE_URL:=@GNOME/glib/$(basename $(PKG_VERSION))
-PKG_HASH:=3652c7f072d7b031a6b5edd623f77ebc5dcd2ae698598abcc89ff39ca75add30
+PKG_SOURCE_URL:=@GNOME/glib/2.70
+PKG_HASH:=200d7df811c5ba634afbf109f14bb40ba7fde670e89389885da14e27c0840742
 
 PKG_MAINTAINER:=Peter Wagner <tripolar@gmx.at>
 PKG_LICENSE:=LGPL-2.1-or-later
@@ -21,13 +21,12 @@
 PKG_CPE_ID:=cpe:/a:gnome:glib
 
 PKG_BUILD_DIR:=$(BUILD_DIR)/glib-$(PKG_VERSION)
+HOST_BUILD_DIR:=$(BUILD_DIR_HOST)/glib-$(PKG_VERSION)
+PKG_BUILD_DEPENDS:=libiconv/host
+HOST_BUILD_DEPENDS:=libiconv/host libffi/host pcre/host
 PKG_CONFIG_DEPENDS:=CONFIG_BUILD_NLS
-PKG_FORTIFY_SOURCE:=0
 PKG_INSTALL:=1
 
-HOST_BUILD_DIR:=$(BUILD_DIR_HOST)/glib-$(PKG_VERSION)
-HOST_BUILD_DEPENDS:=pcre2/host libffi/host
-
 include $(INCLUDE_DIR)/host-build.mk
 include $(INCLUDE_DIR)/package.mk
 include $(INCLUDE_DIR)/nls.mk
@@ -36,7 +35,7 @@
 define Package/glib2
   SECTION:=libs
   CATEGORY:=Libraries
-  DEPENDS:=$(ICONV_DEPENDS) $(INTL_DEPENDS) +zlib +libpthread +libffi +libattr +libpcre2
+  DEPENDS:=$(ICONV_DEPENDS) $(INTL_DEPENDS) +zlib +libpthread +libffi +libattr +libpcre
   TITLE:=glib 2.0
   URL:=http://www.gtk.org/
 endef
@@ -45,10 +44,12 @@
   The GLib library of C routines
 endef
 
+HOST_LDFLAGS += -Wl,-rpath,$(STAGING_DIR_HOSTPKG)/lib
 TARGET_CFLAGS += -ffunction-sections -fdata-sections
-TARGET_LDFLAGS += -Wl,--gc-sections
+TARGET_LDFLAGS += -Wl,--gc-sections $(if $(INTL_FULL),-lintl)
 
 COMP_ARGS= \
+	-Diconv=external \
 	-Dselinux=disabled \
 	-Dlibmount=disabled \
 	-Dman=false \
@@ -58,12 +59,13 @@
 	-Dgtk_doc=false \
 	-Dbsymbolic_functions=true \
 	-Dforce_posix_threads=true \
+	-Dfam=false \
 	-Dtests=false \
 	-Dinstalled_tests=false \
 	-Doss_fuzz=disabled \
 	-Dglib_debug=disabled \
 	-Dglib_assert=false \
-	-Dglib_checks=true \
+	-Dglib_checks=false \
 	-Dlibelf=disabled
 
 MESON_HOST_ARGS += $(COMP_ARGS) -Dxattr=false -Ddefault_library=static -Dnls=disabled
@@ -96,6 +98,10 @@
 		$(1)/usr/lib/
 
 	$(INSTALL_DIR) $(1)/usr/lib/pkgconfig
+ifneq ($(INTL_FULL),)
+	$(SED) '/^Libs:/s/$$$$/ -lintl/' \
+		$(PKG_INSTALL_DIR)/usr/lib/pkgconfig/glib-2.0.pc
+endif
 	$(INSTALL_DATA) \
 		$(PKG_INSTALL_DIR)/usr/lib/pkgconfig/*.pc \
 		$(1)/usr/lib/pkgconfig
--- a/feeds/packages/libs/glib2/patches/006-c99.patch	2024-02-05 13:52:54.217722089 +0200
+++ b/feeds/packages/libs/glib2/patches/006-c99.patch	2024-02-05 14:51:16.944406623 +0200
@@ -1,6 +1,6 @@
 --- a/meson.build
 +++ b/meson.build
-@@ -1045,7 +1045,7 @@ if host_system == 'windows' and (cc.get_
+@@ -949,7 +949,7 @@ if host_system == 'windows' and (cc.get_
    glib_conf.set('HAVE_C99_SNPRINTF', false)
    glib_conf.set('HAVE_C99_VSNPRINTF', false)
    glib_conf.set('HAVE_UNIX98_PRINTF', false)
--- a/feeds/packages/libs/glib2/patches/010-pcre.patch	2024-02-05 13:52:54.217722089 +0200
+++ b/feeds/packages/libs/glib2/patches/010-pcre.patch	1970-01-01 03:00:00.000000000 +0300
@@ -1,10 +0,0 @@
---- a/glib/meson.build
-+++ b/glib/meson.build
-@@ -365,6 +365,7 @@ pcre2_static_args = []
- 
- if use_pcre2_static_flag
-   pcre2_static_args = ['-DPCRE2_STATIC']
-+  pcre2 = pcre2.as_link_whole()
- endif
- 
- glib_c_args = ['-DG_LOG_DOMAIN="GLib"', '-DGLIB_COMPILATION'] + pcre2_static_args + glib_hidden_visibility_args
