--- a/gpio/Makefile
+++ b/gpio/Makefile
@@ -26,6 +26,8 @@
 DESTDIR?=/usr
 PREFIX?=/local
 
+BOARD=$(shell cat ../BOARD)
+
 ifneq ($V,1)
 Q ?= @
 endif
@@ -138,31 +140,6 @@
 tags:	$(SRC)
 	$Q echo [ctags]
 	$Q ctags $(SRC)
-
-.PHONY:	install
-install: gpio
-	$Q echo "[Install]"
-	$Q cp gpio		$(DESTDIR)$(PREFIX)/bin/
-ifneq ($(WIRINGPI_SUID),0)
-	$Q chown root.root	$(DESTDIR)$(PREFIX)/bin/gpio
-	$Q chmod 4755		$(DESTDIR)$(PREFIX)/bin/gpio
-endif
-	$Q mkdir -p		$(DESTDIR)$(PREFIX)/share/man/man1
-	$Q cp gpio.1		$(DESTDIR)$(PREFIX)/share/man/man1/
-
-.PHONY:	install-deb
-install-deb:	gpio
-	$Q echo "[Install: deb]"
-	$Q install -m 0755 -d							$(CURDIR)/../debian-template/wiringPi/usr/bin
-	$Q install -m 0755 gpio							$(CURDIR)/../debian-template/wiringPi/usr/bin
-	$Q install -m 0755 -d							$(CURDIR)/../debian-template/wiringPi/usr/share/man/man1
-	$Q install -m 0644 gpio.1						$(CURDIR)/../debian-template/wiringPi/usr/share/man/man1
-
-.PHONY:	uninstall
-uninstall:
-	$Q echo "[UnInstall]"
-	$Q rm -f $(DESTDIR)$(PREFIX)/bin/gpio
-	$Q rm -f $(DESTDIR)$(PREFIX)/share/man/man1/gpio.1
 
 .PHONY:	depend
 depend:
