--- a/gpio/OrangePi.c
+++ b/gpio/OrangePi.c
@@ -493,8 +499,8 @@
   12, 13,   //21, 22
   14, 15,   //23, 24
   -1, 16,   //25, 26
+  17, -1,   //27, 28 //  // +  17, 18, //27, 28 wpi 17-18
 
-  -1, -1,   //27, 28
   -1, -1,   //29, 30
   -1, -1,   //31, 32
   -1, -1,   //33, 34
@@ -509,21 +519,21 @@
 {
       NULL,
 
- "    3.3V", "5V      ",
- "   SDA.0", "5V      ",
- "   SCL.0", "GND     ",
- "     PA6", "TXD.1   ",
- "     GND", "RXD.1   ",
- "   RXD.2", "PA07    ",
- "   TXD.2", "GND     ",
- "   CTS.2", "SDA.1   ",
- "    3.3V", "SCK.1   ",
- "  MOSI.1", "GND     ",
- "  MISO.1", "RTS.2   ",
- "  SCLK.1", "CE.1    ",
- "     GND", "PA10    ",
-
-       NULL, NULL,
+ " +3.3v  ", "  +5v   ",
+ "PA12 sda", "  +5v   ",
+ "PA11 scl", "   GND  ",
+ "    PA6 ", " PG6    ",
+ "  GND   ", " PG7    ",
+ "    PA1 ", " PA7    ",
+ "    PA0 ", "   GND  ",
+ "    PA3 ", " PA19   ",
+ " +3.3v  ", " PA18   ",
+ "    PA15", "   GND  ",
+ "    PA16", " PA2    ",
+ "    PA14", " PA13   ",
+ "  GND   ", " PA10   ",
+ " RED-LED", " PWR-LED",
+
        NULL, NULL,
        NULL, NULL,
        NULL, NULL,
@@ -1086,6 +1090,8 @@
 void OrangePiReadAll(void)
 {
     int pin;
+    char name [80] ;
+    FILE *fd ;
 	int tmp = wiringPiDebug;
     wiringPiDebug = FALSE;
 
@@ -1089,39 +1093,13 @@
 	int tmp = wiringPiDebug;
     wiringPiDebug = FALSE;
 
+    if ((fd = fopen ("/proc/device-tree/model", "r")) != NULL)
+    {
+      if ( fgets (name, 80, fd) ){;}
+      fclose (fd) ;
+    }
+
-#ifdef CONFIG_ORANGEPI_PC2
-    printf (" +------+-----+----------+------+---+  OPi PC2 +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_PRIME
-    printf (" +------+-----+----------+------+---+   PRIME  +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_ZEROPLUS
-    printf (" +------+-----+----------+------+---+ ZEROPLUS +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_ZEROPLUS2_H5
-    printf (" +------+-----+----------+------+---+ZEROPLUS 2+---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_2G_IOT
-    printf (" +------+-----+----------+------+---+  2G-IOT  +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_H3
-    printf (" +------+-----+----------+------+---+OrangePiH3+---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_ZEROPLUS2_H3
-    printf (" +------+-----+----------+------+---+ZEROPLUS 2+---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_ZERO
-    printf (" +------+-----+----------+------+---+  OPi H2  +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_WIN
-    printf (" +------+-----+----------+------+---+ OPi Win  +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_LITE2 
-    printf (" +------+-----+----------+------+---+  OPi H6  +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_3
-    printf (" +------+-----+----------+------+---+   OPi 3  +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_ZERO2
-    printf (" +------+-----+----------+------+---+  Zero 2  +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_RK3399
-    printf (" +------+-----+----------+------+---+OPi RK3399+---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_4
-    printf (" +------+-----+----------+------+---+OrangePi 4+---+---+--+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_800
-    printf (" +------+-----+----------+------+---+   800    +---+---+--+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_R1PLUS
-    printf (" +------+-----+----------+------+---+ R1 Plus  +---+---+--+----------+-----+------+\n");
-#endif
+    printf ("\n\n +------+-----+----------+- %s -+----------+-----+------+\n", name);
 
     printf (" | GPIO | wPi |   Name   | Mode | V | Physical | V | Mode | Name     | wPi | GPIO |\n");
     printf (" +------+-----+----------+------+---+----++----+---+------+----------+-----+------+\n");
@@ -1140,42 +1183,13 @@
 #endif
         readallPhys(pin);
 
+    printf (" +------+-----+----------+------+---+-- LEDs --+---+------+----------+-----+------+\n");
+    readallPhys (27) ;
+
     printf (" +------+-----+----------+------+---+----++----+---+------+----------+-----+------+\n");
     printf (" | GPIO | wPi |   Name   | Mode | V | Physical | V | Mode | Name     | wPi | GPIO |\n");
 
-#ifdef CONFIG_ORANGEPI_PC2
-    printf (" +------+-----+----------+------+---+  OPi PC2 +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_PRIME
-    printf (" +------+-----+----------+------+---+   PRIME  +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_ZEROPLUS
-    printf (" +------+-----+----------+------+---+ ZEROPLUS +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_ZEROPLUS2_H5
-    printf (" +------+-----+----------+------+---+ZEROPLUS 2+---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_2G_IOT
-    printf (" +------+-----+----------+------+---+  2G-IOT  +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_H3
-    printf (" +------+-----+----------+------+---+OrangePiH3+---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_ZEROPLUS2_H3
-    printf (" +------+-----+----------+------+---+ZEROPLUS 2+---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_ZERO
-    printf (" +------+-----+----------+------+---+  OPi H2  +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_WIN
-    printf (" +------+-----+----------+------+---+ OPi Win  +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_LITE2 
-    printf (" +------+-----+----------+------+---+  OPi H6  +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_3
-    printf (" +------+-----+----------+------+---+   OPi 3  +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_ZERO2
-    printf (" +------+-----+----------+------+---+  Zero 2  +---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_RK3399
-    printf (" +------+-----+----------+------+---+OPi RK3399+---+------+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_4
-    printf (" +------+-----+----------+------+---+OrangePi 4+---+---+--+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_800
-    printf (" +------+-----+----------+------+---+   800    +---+---+--+----------+-----+------+\n");
-#elif CONFIG_ORANGEPI_R1PLUS
-    printf (" +------+-----+----------+------+---+ R1 Plus  +---+---+--+----------+-----+------+\n");
-#endif
+    printf (" +------+-----+----------+- %s -+----------+-----+------+\n\n", name);
 
     wiringPiDebug = tmp;
 }
