--- a/wiringPiD/Makefile
+++ b/wiringPiD/Makefile
@@ -64,30 +70,6 @@
 tags:	$(SRC)
 	$Q echo [ctags]
 	$Q ctags $(SRC)
-
-.PHONY:	install
-install: wiringpid
-	$Q echo "[Install]"
-	$Q mkdir -p		$(DESTDIR)$(PREFIX)/sbin
-	$Q cp wiringpid		$(DESTDIR)$(PREFIX)/sbin/
-	$Q chown root.root	$(DESTDIR)$(PREFIX)/sbin/wiringpid
-
-#	$Q mkdir -p		$(DESTDIR)$(PREFIX)/man/man8
-#	$Q cp gpio.1		$(DESTDIR)$(PREFIX)/man/man8/
-
-.PHONY:	install-deb
-install-deb:	gpio
-	$Q echo "[Install: deb]"
-	$Q install -m 0755 -d							$(CURDIR)/../debian-template/wiringPi/usr/bin
-	$Q install -m 0755 gpio							$(CURDIR)/../debian-template/wiringPi/usr/bin
-	$Q install -m 0755 -d							$(CURDIR)/../debian-template/wiringPi/man/man1
-	$Q install -m 0644 gpio.1						$(CURDIR)/../debian-template/wiringPi/man/man1
-
-.PHONY:	uninstall
-uninstall:
-	$Q echo "[UnInstall]"
-	$Q rm -f $(DESTDIR)$(PREFIX)/sbin/wiringpid
-	$Q rm -f $(DESTDIR)$(PREFIX)/man/man8/wiringpid.8
 
 .PHONY:	depend
 depend:
