--- a/main.c
+++ b/main.c
@@ -502,7 +510,7 @@
 	int if_id;
 	struct ieee80211_hw *dev;
 	struct xradio_common *hw_priv;
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 13, 0)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0))
 	unsigned char addr[ETH_ALEN];
 #else
 	unsigned char randomaddr[ETH_ALEN];
@@ -520,7 +530,7 @@
 	hw_priv->sdio_func = func;
 	sdio_set_drvdata(func, hw_priv);
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 13, 0)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0))
 	// fill in mac addresses
 	if (hw_priv->pdev->of_node) {
 		err = of_get_mac_address(hw_priv->pdev->of_node, addr);
